pkg/runtime/engines/singularity/config/config.go

```diff
@@ -485,3 +489,98 @@ func (e *EngineConfig) GetDeleteImage() bool {
 func (e *EngineConfig) SetDeleteImage(delete bool) {
 	e.JSON.DeleteImage = delete
 }
+
+// GetPluginConfig retrieves the configuration for the named plugin
+func (e *EngineConfig) GetPluginConfig(plugin string, cfg interface{}) error {
+	if tmp, found := e.Plugin[plugin]; found {
+		return json.Unmarshal(tmp, cfg)
+	}
+
+	return nil
+}
+
+// SetPluginConfig sets the configuration for the named plugin
+func (e *EngineConfig) SetPluginConfig(plugin string, cfg interface{}) error {
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+	e.Plugin[plugin] = json.RawMessage(data)
+	return nil
+}
+
+// GetPluginFuseMounts returns the list of plugins which have a valid
+// FUSE configuration.
+//
+// In this context "valid" means that both the mount point and the FUSE
+// driver program exist in the configuration. This function DOES NOT
+// check if the /dev/fuse file descriptor has been assigned.
+func (e *EngineConfig) GetPluginFuseMounts() []string {
+	var list []string
+
+	for name, raw := range e.Plugin {
+		var info struct {
+			Fuse struct {
+				DevFuseFd  int
+				MountPoint string
+				Program    []string
+			}
+		}
+		if err := json.Unmarshal(raw, &info); err != nil {
+			// do not log anything because this would
+			// introduce a lot of noise into the log, even
+			// for the debug level
+			continue
+		}
+		if len(info.Fuse.Program) > 0 && info.Fuse.MountPoint != "" {
+			// This a valid configuration
+			list = append(list, name)
+		}
+	}
+
+	return list
+}
+
+// SetPluginFuseFd sets the /dev/fuse file descriptor fd for the
+// specified plugin
+//
+// This function tries to make sure that any additional configuration
+// already found in the "Fuse" object is preserved.
+func (e *EngineConfig) SetPluginFuseFd(name string, fd int) error {
+	raw, found := e.Plugin[name]
+	if !found {
+		// named plugin does not have a configuration
+		// entry, skip
```

Comment:
Why are these comments indicating that these calls are skipped when `openDevFuse()` is catching and stopping iteration over plugin fuse mounts if an error is encountered

Reply:
LOL... that's fair.

I mean "skip" in the sense that this won't be doing anything with this plugin. The "skip" part was probably more accurate in the original context where this code existed. I'll fix this. Thanks.